crates/matrix-sdk-base/src/client.rs

```diff
@@ -875,9 +875,19 @@ impl BaseClient {
                 )
                 .await?;
 
+            // Save the new `RoomInfo`.
+            changes.add_room(room_info);
+
             self.handle_room_account_data(&room_id, &new_info.account_data.events, &mut changes)
                 .await;
 
+            // `Self::handle_room_account_data` might have updated the `RoomInfo`. Let's
+            // fetch it again.
+            //
+            // SAFETY: `unwrap` is safe because the `RoomInfo` has been inserted 2 lines
+            // above.
+            let mut room_info = changes.room_infos.get(&room_id).unwrap().clone();
```

Comment:
So the way it's done in the sliding-sync variant of this function, is that we edit the `room_info`, and don't mess at all with the `changes` until the end of the function where we put the room info into the changes. We should do that here too, for consistency at least. I'll write a followup.